### PR TITLE
8323605: Java source launcher should not require `--source ...` to enable preview

### PR DESCRIPTION
**Pull request by sormuras**
Please review this PR targeting the source launcher that instead of throwing an error reading _"--enable-preview must be used with --source"_ adds the required compiler arguments on-the-fly.

The add arguments are `--release` and `feature`, with `feature` being the release feature number of the current Java runtime as acquired by `Runtime.version().feature()`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8325337](https://bugs.openjdk.org/browse/JDK-8325337) to be approved

### Issues
 * [JDK-8323605](https://bugs.openjdk.org/browse/JDK-8323605): Java source launcher should not require `--source ...` to enable preview (**Enhancement** - P4)
 * [JDK-8325337](https://bugs.openjdk.org/browse/JDK-8325337): Java source launcher should not require `--source ...` to enable preview (**CSR**)

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**) ⚠️ Review applies to [23fba2bf](https://git.openjdk.org/jdk/pull/17731/files/23fba2bf31bf15b4002421b49eb7b24d5d0ec928)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17731/head:pull/17731` \
`$ git checkout pull/17731`

Update a local copy of the PR: \
`$ git checkout pull/17731` \
`$ git pull https://git.openjdk.org/jdk.git pull/17731/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17731`

View PR using the GUI difftool: \
`$ git pr show -t 17731`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17731.diff">https://git.openjdk.org/jdk/pull/17731.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17731#issuecomment-1929383084)

### CODE REVIEW

**Review by mcimadamore — APPROVED** [review]
Looks good!

File: src/java.base/share/man/java.1 [hunk 1]
```diff
@@ -279,6 +279,12 @@ the compilation.
 This sets both the source version accepted by compiler and the system
 API that may be used by the code in the source file.
 .IP \[bu] 2
+If \f[V]--enable-preview\f[R] is specified, the \f[V]--source N\f[R]
+arguments can be omitted.
+The Java Runtime of version \f[V]N\f[R] implies an implicit
```
Comment (dholmes-ora) [inline src/java.base/share/man/java.1:284]: I think that should be `A Java runtime version ...`

Comment (dholmes-ora) [reply src/java.base/share/man/java.1:284]: Also `implies an implicit` is doubling up. I suggest the following complete sentence:
```
If the Java runtime version is N, then `--release N` is implied when compiling source files.
```

Comment (sormuras) [reply src/java.base/share/man/java.1]: Done via https://github.com/openjdk/jdk/pull/17731/commits/a82e4fe3d05a411635bbce0a69e17218184e7ab6

**Review by dholmes-ora — APPROVED** [review]
manpage update is good. Thanks

**Pull request closed without merging**